test/jdk/java/lang/reflect/Proxy/DefaultMethods.java

```diff
@@ -322,33 +322,31 @@ public void testIllegalArgument(Class<?>[] interfaces, Class<?> defc, String nam
         }
     }
 
-    @DataProvider(name = "arguments")
-    private Object[][] arguments() {
+    @DataProvider(name = "illegalArguments")
+    private Object[][] illegalArguments() {
         return new Object[][] {
-                new Object[] {},
-                new Object[] { 100, "foo", 100 },
-                new Object[] { "foo", 100},
+            new Object[] {},
+            new Object[] { 100 },
+            new Object[] { 100, "foo", 100 },
+            new Object[] { 100L, "foo" },
+            new Object[] { "foo", 100},
+            new Object[] { null, "foo" }
         };
     }
 
-    @Test(dataProvider = "arguments", expectedExceptions = {IllegalArgumentException.class})
-    public void testWrongArguments(Object... args) throws Exception {
+    @Test(dataProvider = "illegalArguments", expectedExceptions = {IllegalArgumentException.class})
+    public void testIllegalArguments(Object... args) throws Exception {
         ClassLoader loader = DefaultMethods.class.getClassLoader();
         I4 proxy = (I4)Proxy.newProxyInstance(loader, new Class<?>[]{I4.class}, HANDLER);
         Method m = I4.class.getMethod("mix", int.class, String.class);
         assertTrue(m.isDefault());
+        if (args.length == 0) {
+            // substitute empty args with null since @DataProvider doesn't allow null array
+            args = null;
+        }
         Proxy.invokeDefaultMethod(proxy, m, args);
     }
 
-    @Test(expectedExceptions = {IllegalArgumentException.class})
-    public void testNullArguments() throws Exception {
-        ClassLoader loader = DefaultMethods.class.getClassLoader();
-        I4 proxy = (I4)Proxy.newProxyInstance(loader, new Class<?>[]{I4.class}, HANDLER);
-        Method m = I4.class.getMethod("mix", int.class, String.class);
-        assertTrue(m.isDefault());
-        Proxy.invokeDefaultMethod(proxy, m, (Object[])null);
-    }
-
```

Comment:
Do you mind adding a new test case to check if NPE thrown by the method body resulting in `InvocationTargetException`?   I missed that case.

Reply:
Ok.